### PR TITLE
feat: checkout to the ref instead of FETCH_HEAD if ref is a branch

### PR DESCRIPTION
**Pull request by dgozalo**
Signed-off-by: Daniel Gozalo <dgozalo@cloudbees.com>

#### Submitter checklist

- [X] Change is code complete and matches issue description.
- [X] Change is covered by existing or new tests.

#### Description
We need to be able to use a branch as a versions stream ref in order to get our automated release system to work, if the ref is a branch it will look for it in the remote branches and then checkout to it if it is.

#### Which issue this PR fixes

fixes #

### CODE REVIEW

File: pkg/versionstream/versionstreamrepo/gitrepo.go [hunk 1]
```diff
@@ -238,3 +253,16 @@ func resolveRefToTag(dir string, commitish string, gitter gits.Gitter) (string,
 	}
 	return resolved, nil
 }
+
+func isBranch(dir string, ref string, gitter gits.Gitter) (bool, error) {
```
Comment (abayer) [inline pkg/versionstream/versionstreamrepo/gitrepo.go]: This is useful enough that I think it would make sense to go in the git helpers.

Comment (abayer) [reply pkg/versionstream/versionstreamrepo/gitrepo.go]: https://github.com/jenkins-x/jx/blob/master/pkg/gits/helpers.go specifically =)

Comment (dgozalo) [reply pkg/versionstream/versionstreamrepo/gitrepo.go]: Done

**Pull request merged**